platforms/posix/CMakeLists.txt

```diff
@@ -92,6 +92,77 @@ endif()
 # install
 #
 
-install(TARGETS px4 DESTINATION ${CMAKE_INSTALL_BINDIR})
-install(DIRECTORY ${PROJECT_SOURCE_DIR}/ROMFS DESTINATION ${CMAKE_INSTALL_DATADIR}/${PROJECT_NAME})
-install(DIRECTORY ${PROJECT_SOURCE_DIR}/posix-configs DESTINATION ${CMAKE_INSTALL_DATADIR}/${PROJECT_NAME})
+# px4 app
+install(
+	TARGETS
+		px4
+	DESTINATION
+		${CMAKE_INSTALL_BINDIR}
+	)
+
+# px4 dirs
+install(
+	DIRECTORY
+		${PROJECT_SOURCE_DIR}/integrationtests
+		${PROJECT_SOURCE_DIR}/launch
+		${PROJECT_SOURCE_DIR}/posix-configs
+		${PROJECT_SOURCE_DIR}/ROMFS
+		${PROJECT_SOURCE_DIR}/test
+	DESTINATION
+		${CMAKE_INSTALL_DATADIR}/${PROJECT_NAME}
+	)
+
+# px4 files
+install(
+	FILES
+		${PROJECT_SOURCE_DIR}/CMakeLists.txt
+		${PROJECT_SOURCE_DIR}/package.xml
+	DESTINATION
+		${CMAKE_INSTALL_DATADIR}/${PROJECT_NAME}
+	)
+
+# px4 Tools dirs
+install(
+	DIRECTORY
+		${PROJECT_SOURCE_DIR}/Tools/ecl_ekf
+	DESTINATION
+		${CMAKE_INSTALL_DATADIR}/${PROJECT_NAME}/Tools
+	)
+
+# px4 Tools files
+install(
+	FILES
+		${PROJECT_SOURCE_DIR}/Tools/setup_gazebo.bash
+		${PROJECT_SOURCE_DIR}/Tools/upload_log.py
+	DESTINATION
+		${CMAKE_INSTALL_DATADIR}/${PROJECT_NAME}/Tools
+	)
+
+# sitl_gazebo built plugins
+install(
+	DIRECTORY
+		${PROJECT_SOURCE_DIR}/build/posix_sitl_default/build_gazebo
+	DESTINATION
+		${CMAKE_INSTALL_DATADIR}/${PROJECT_NAME}/build/posix_sitl_default
```

Comment:
I need to think about how to do this independant of directory.

Reply:
We could create a list of the files and use `DIRECTORY` with `FILES_MATCHING` to reduce the number of `install` commands. I originally did it this way to be clear about where everything is coming from while I was determining the pieces that we needed. I'm not attached to this implementation.